src/onemod/fsutils/io.py

```diff
@@ -0,0 +1,227 @@
+import json
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Literal, Type
+
+import dill
+import pandas as pd
+import polars as pl
+import tomli
+import tomli_w
+import yaml
+
+
+class DataIO(ABC):
```

Comment:
As we discussed, one thing we can consider is to use `FileIO` as the name, and only provide `csv, parquet` and other potential tabular data format with different backend options :)

Reply:
Yes I think this is a good idea. I discovered some additional complexity though I think, so do you mind if I make a new ticket for this? Don't think I'll be able to quite get to it this week.

Reply:
Yes! That sounds good! This is not critical at all, just nerding out on the naming :)